contracts/PetPark.sol

```diff
@@ -1,7 +1,100 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
 
-
 contract PetPark {
+    enum AnimalType {
+        None,
+        Fish,
+        Cat,
+        Dog,
+        Rabbit,
+        Parrot
+    }
+
+    enum Gender {
+        Male,
+        Female
+    }
+
+    struct Borrower {
+        uint age;
+        uint gender;
+        uint animalTypeIdx;
+    }
+
+    mapping(uint => uint) animalCountMap;
+    mapping(address => Borrower) borrowers;
+
+    address private owner;
+
+    event Added(uint, uint);
+    event Borrowed(uint);
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    function add(uint animalTypeIdx, uint count) external {
+        require(msg.sender == owner, "Not an owner");
+        require(animalTypeIdx > 0 && animalTypeIdx < 6, "Invalid animal");
+        animalCountMap[animalTypeIdx] += count;
+        emit Added(animalTypeIdx, count);
+    }
+
+    function borrow(
+        uint age,
+        uint gender,
+        uint animalTypeIdx
+    ) external {
+        require(age > 0, "Invalid Age");
+        require(animalTypeIdx != 0, "Invalid animal type");
+        require(
+            animalCountMap[animalTypeIdx] > 0,
+            "Selected animal not available"
+        );
+
+        Borrower storage borrower = borrowers[msg.sender];
```

Comment:
When you only need to read from a struct instance, it is better to store it in memory